### PR TITLE
FIG-30313: Create Radio component

### PR DESCRIPTION
**Pull request by ailincaibianca**
[FIG-30313](https://digital-science.atlassian.net/browse/FIG-30313)

https://github.com/figshare/fcl/assets/63391401/a54d68ef-1099-4f61-8e19-cb8829dcaab8

[FIG-30313]: https://digital-science.atlassian.net/browse/FIG-30313?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AlexBabusanu — APPROVED** [review]

File: packages/ui/input/radio/index.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import classnames from "classnames";
+import { bool, func, string } from "prop-types";
+
+import styles from "./index.css";
+
+
+export default function Radio({ checked, className, disabled, error, label, onChange, value: inputValue }) {
```
Comment (antalutrazvan) [inline packages/ui/input/radio/index.jsx:8]: @ailincaibianca destructure `{ ...other}` from the `props` object to the `<input` node. There are properties of importance to a `input type="radio"` to pass. Such as `name`.

Also, `id` sent to `<input` should not only be: `inputValue`.

Consider if the value is `1` for one `<Radio` but then another `<Radio` has the same value of `1` somewhere else in the app.

I would suggest we set it as either an explicitly passed `id` through props. or, if it is not passed, a combination of `name + value`:
```
 export default function Radio({ (...), id, name = "", ...others }) {
 const inputId = id ? id : `${name}-${inputValue}`;
 
```

File: packages/ui/input/radio/index.css [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+.error input[type="radio"] {
+  border: 1px solid var(--color-typography-error);
+}
+
+.error input[type="radio"]::before {
+  box-shadow: inset calc(2 * var(--gridSize)) calc(2 * var(--gridSize))  var(--color-typography-error);
+}
+
+input[type="radio"]:disabled {
+  border: 1px solid #bbb;
```
Comment (andreistangacianu) [inline packages/ui/input/radio/index.css:78]: this should be something from colors

File: packages/ui/input/radio/index.css [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+input[type="radio"]:disabled + label {
+  color: #bbb;
+}
+
+input[type="radio"][disabled]::before {
+  box-shadow: inset calc(2 * var(--gridSize)) calc(2 * var(--gridSize)) #bbb;
```
Comment (andreistangacianu) [inline packages/ui/input/radio/index.css:86]: color plz

File: packages/ui/input/radio/index.jsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    onClickRightIcon,
+    value: inputValue,
+    ...others
+  }
+) {
+  const inputId = id ? id : `${name}-${inputValue}`;
+  const handleChange = (event) => {
+    const { value } = event.target;
+
+    if (onChange) {
```
Comment (andreistangacianu) [inline packages/ui/input/radio/index.jsx:31]: let's test `typeof onChange === function` - it's sfer

File: packages/ui/input/radio/index.jsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    name,
+    onChange,
+    onClickRightIcon,
+    value: inputValue,
+    ...others
+  }
+) {
+  const inputId = id ? id : `${name}-${inputValue}`;
+  const handleChange = (event) => {
+    const { value } = event.target;
```
Comment (andreistangacianu) [inline packages/ui/input/radio/index.jsx:29]: will `event.target` always exist ?

**Comment by andreistangacianu** [comment]
Looks awesome, thank you!

**Review by antalutrazvan — CHANGES_REQUESTED** [review]

File: packages/ui/input/radio/index.jsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    value: inputValue,
+    ...others
+  }
+) {
+  const inputId = id || `${name}-${inputValue}`;
+  const handleChange = (event) => {
+    const { value } = event?.target;
+
+    if (typeof onChange === "function") {
+      onChange(value);
```
[the remark this answers is not in the document]
Comment (ailincaibianca) [reply packages/ui/input/radio/index.jsx]: Fixed

**Review by antalutrazvan — APPROVED** [review]

**Pull request merged**